sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/GroupByDocumentQueryExecutionContext.java

```diff
@@ -67,32 +68,49 @@
     @Override
     public Flux<FeedResponse<T>> drainAsync(int maxPageSize) {
         return this.component.drainAsync(maxPageSize)
-                   .collectList()
-                   .map(superList -> {
-                       double requestCharge = 0;
-                       HashMap<String, String> headers = new HashMap<>();
-                       List<Document> documentList = new ArrayList<>();
-                       /* Do groupby stuff here */
-                       // Stage 1:
-                       // Drain the groupings fully from all continuation and all partitions
-                       for (FeedResponse<T> page : superList) {
-                           List<Document> results = (List<Document>) page.getResults();
-                           documentList.addAll(results);
-                       }
-
-                       this.aggregateGroupings(documentList);
-
-                       // Stage 2:
-                       // Emit the results from the grouping table page by page
-
-                       List<Document> groupByResults = this.groupingTable.drain(maxPageSize);
-
-                       headers.put(HttpConstants.HttpHeaders.REQUEST_CHARGE, Double.toString(requestCharge));
-                       FeedResponse<Document> frp =
-                           BridgeInternal.createFeedResponse(groupByResults, headers);
-
-                       return (FeedResponse<T>) frp;
-                   }).flux();
+            .collectList()
+            .map(superList -> {
+                double requestCharge = 0;
+                HashMap<String, String> headers = new HashMap<>();
+                List<Document> documentList = new ArrayList<>();
+                /* Do groupBy stuff here */
+                // Stage 1:
+                // Drain the groupings fully from all continuation and all partitions
+                for (FeedResponse<T> page : superList) {
+                    List<Document> results = (List<Document>) page.getResults();
+                    documentList.addAll(results);
+                    requestCharge += page.getRequestCharge();
+                }
+
+                this.aggregateGroupings(documentList);
+
+                // Stage 2:
+                // Emit the results from the grouping table page by page
+                return (FeedResponse<T>)createFeedResponseFromGroupingTable(maxPageSize, requestCharge);
+            }).expand(tFeedResponse -> {
+                // Emit the results from the grouping table page by page
```

Comment:
Please add some comments to expand (explaining how it will be called second time onwards)

Reply:
Updated